### ISSUE TITLE
[BUG] Accessibility Checker plugins generates incorrect violation

### ISSUE THREAD

**Issue by teamacs** [IBMa/equal-access#248]
## What tool(s) are you using?

- [ ] `accessibility-checker-engine`
- [x] `accessibility-checker`
- [ ] `karma-accessibility-checker`
- [ ] `rule-server`
- [ ] `boilerplates`
- [ ] `other`

## Detailed description
Whenever you run the Accessibility Checker plugin (Firefox and Chrome), the plugin will generate violations of `Form control element <button> has no associated label`. One example is for the following piece of html:

`<button title="Close store" class="tabClose" _ngcontent-tjd-c76="">`
`<span><i class="fa fa-times" aria-hidden="true"></i></span>`
`</button>`

In the "What to do" section it describes four fixes:
>  Add a for attribute to a visible label with the id of this control (e.g. <label for="controlID">);
    OR put the UI element inside a visible <label> element;
    OR add an aria-labelledby attribute to the control. It must point to visible text on the page that is meaningful as a label;
    OR only if the design cannot have a visible label, use the title attribute to provide a label.

<img width="1681" alt="Screen Shot 2020-10-06 at 3 14 15 PM" src="https://user-images.githubusercontent.com/42381766/95257863-8fb9eb00-07ea-11eb-8085-ac5b6c6ea6e8.png">

Based on the above "What to do", the above piece of html should not generate a violation because it contains the title attribute. In a little testing, I was able to get the violation to go away if there was actual text in the button and not just the icon. I don't believe that the icon should be a problem as that is another reason for having the title text.

**Comment by tombrunet** [comment]
DUPE with https://github.com/IBMa/equal-access/issues/238 . Fix is currently available in the `preview` archive. It will be in the next rule pack. We expect that to deploy this week.

**Comment by teamacs** [comment]
A new rule set showed up today even though its dated last week. The issue still persists.

<img width="978" alt="Screen Shot 2020-10-16 at 8 53 52 AM" src="https://user-images.githubusercontent.com/42381766/96267325-49ece780-0f8d-11eb-9394-946e00786aa7.png">
<img width="1680" alt="Screen Shot 2020-10-16 at 8 53 00 AM" src="https://user-images.githubusercontent.com/42381766/96267330-4c4f4180-0f8d-11eb-851c-69c0722fbb65.png">

**Comment by tombrunet** [comment]
@teamacs Can you close your browser and then try again? The old server hadn't shutdown, so it was randomly pulling the old rules.

**Comment by teamacs** [comment]
Just did so and still having the issue. Plugin version is still 3.1.0 and rule set is still 07 October 2020 - Latest Deployment. Using Firefox Developer Edition v83.0b1 with a clean cache.

**Comment by tombrunet** [comment]
Confirmed that this is still failing.

**Comment by tombrunet** [comment]
Unable to reproduce in current build.
